app/models/book.rb

```diff
@@ -131,55 +129,31 @@ def price
   end
 
   def as_json(options={})
-    super(
-      methods: [
-        :genre_name,
-        :subcategory_name,
-        :language_name,
-        :country_name,
-        :levels_name,
-        :publisher_name,
-        :authors_name,
-        :update_status,
-        :donated?,
-        :updated_date,
-        :url
-      ]
-    )
+    options[:methods] = [:subcategory_name, :update_status, :donated?, :updated_date, :url]
+    super(options)
+  end
+
+  def as_indexed_json(options={})
+    as_json({
+      include: {
+        authors: {only: :name},
+        country: {only: :name},
+        genre: {only: :name},
+        language: {only: :name},
+        levels: {only: :name},
+        publisher: {only: :name}
+      }
+    })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.